GAE/src/org/waterforpeople/mapping/app/web/DeviceNotificationRestServlet.java

```diff
@@ -54,20 +58,20 @@ protected RestRequest convertRequest() throws Exception {
     protected RestResponse handleRequest(RestRequest req) throws Exception {
         DeviceNotificationRequest dnReq = (DeviceNotificationRequest) req;
         DeviceNotificationResponse resp = new DeviceNotificationResponse();
-        Device d = getDevice(dnReq);
 
-        if (d == null) {
-            return resp;
+        Device d = getDevice(dnReq);
+        if (d != null) {
+            DeviceFileJobQueueDAO jobDao = new DeviceFileJobQueueDAO();
+    
+            List<DeviceFileJobQueue> missingByDevice = jobDao.listByDeviceId(d
+                    .getKey().getId());
+            List<DeviceFileJobQueue> missingUnknown = jobDao.listByUnknownDevice();
+    
+            resp.setMissingFiles(missingByDevice);
+            resp.setMissingUnknown(missingUnknown);
         }
-
-        DeviceFileJobQueueDAO jobDao = new DeviceFileJobQueueDAO();
-
-        List<DeviceFileJobQueue> missingByDevice = jobDao.listByDeviceId(d
-                .getKey().getId());
-        List<DeviceFileJobQueue> missingUnknown = jobDao.listByUnknownDevice();
-
-        resp.setMissingFiles(missingByDevice);
-        resp.setMissingUnknown(missingUnknown);
+        
+        resp.setDeletedSurvey(getDeletedSurveys(dnReq));
 
```

Comment:
Is there any particular reason this is outside of the `if (d != null) {}` block?